### PR TITLE
feat: Upload File, Tests, Restructure

### PR DESCRIPTION
**Pull request by Woynert**
## Includes

- Upload File implementation
- Soap field validation with Jakarta
- Reorganize SOAP controllers
- Tests covering files related to Upload File `> 80%`
- Updated Open API Specification
- Added further testing commands in README

## How to try

1. Start the docker compose which will start the following services:

```sh
docker-compose up
```

| | |
|---|---|
|Authentication|`127.0.0.1:8081`|
|Metadata|`127.0.0.1:8082`|
|Worker|`127.0.0.1:1099`|
|Postgres-db|`127.0.0.1:5432`|
|adminer|`127.0.0.1:5050` <- recommended for viewing tables|

2. Then run gateway tests: All test should pass.

```sh
./gradlew cleanTest testCodeCoverageReport
```

3. Check there are files uploaded in the worker service. (You can check the worker id with `docker ps`):

```sh
docker exec <WORKER CONTAINER ID> sh -c 'find /var/capy/ | sed -e "s/[^-][^\/]*\//   │/g" -e "s/│\([^ ]\)/├── \1/"'
```

```
   │   │
   │   ├── store
   │   │   ├── backups
   │   │   │   ├── volume3
   │   │   │   ├── volume2
   │   │   │   │   ├── 0edf2712-67df-4f06-aa0c-f130ef8adf15
   │   │   │   │   ├── 8427b18d-fec7-4e69-88be-226a88754ef9
   │   │   │   ├── volume1
   │   │   ├── files
   │   │   │   ├── volume3
   │   │   │   ├── volume2
   │   │   │   │   ├── 0edf2712-67df-4f06-aa0c-f130ef8adf15
   │   │   │   │   ├── 8427b18d-fec7-4e69-88be-226a88754ef9
   │   │   │   ├── volume1
```

4. Also, check the metadata tables with adminer http://127.0.0.1:5050.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    restart: on-failure
+    ports:
+      - "127.0.0.1:5432:5432"
+    environment:
+      POSTGRES_MULTIPLE_DATABASES: authdb,metadatadb
+      POSTGRES_PASSWORD: password
+      POSTGRES_USER: username
+    volumes:
+      - ./.docker-psql-mult-db:/docker-entrypoint-initdb.d/init.sh
+      - ../docker-mounts/capydb:/var/lib/postgresql/data
```
Comment (shoriwe) [inline docker-compose.yml:66]: Do not mount the volume on which database will be stored, we have no need to maintain persistence

File: soap-python-client.py [hunk 2]
```diff
@@ -23,3 +25,15 @@
...
 print(result)
+
+# upload demo
+with open ("./LICENSE", "rb") as in_file:
+
+    result = client.service.file_upload({
+        'fileName': f"karolsticker{random.random()}.png",
+        'fileContent': in_file.read(),
+        'location': 'None',
+        'token': 'eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2OTgwNTc1MTYsInV1aWQiOiIyNGVlOWU4Ni0zNmFjLTQyNWMtYWU4Mi04Nzk3MjFmODI5NTMifQ.0IlrR1KTNa_SnJwvDsf-q9TkmP_DrEaPCP-1xVVSsQArL3-AHjp0YJETaS4sQjSNtkzhN2Fz0WkEqV-TnDDXCQ'
```
Comment (shoriwe) [inline soap-python-client.py:36]: Does this is deterministic? For example running the client script in a fresh environment will successfully pass all its tests?

Comment (Woynert) [reply soap-python-client.py:36]: [soap-python-client.py](https://github.com/hawks-atlanta/gateway-java/pull/42/files/d809f030b14f626212a33c96cc294199bee9074d#diff-183c06352659a8e335d564adda4092cd120cca9010b3b1d67f182fbeae1760a6) is not a test. It's a file to help developers debug their SOAP methods. Only for reference, not deterministic, those methods will mostly fail.

Do you think we should keep it?

Comment (shoriwe) [reply soap-python-client.py:36]: Then ignore this comment

**Review by shoriwe — APPROVED** [review]
Squash & Merge

**Pull request merged**